### ISSUE TITLE
MigrationFailedException - Hint: Must be superuser to create this extension. [Failed SQL: CREATE EXTENSION IF NOT EXISTS "uuid-ossp"]

### ISSUE THREAD

**Issue by uweschaefer** [factcast/factcast#372]
_From @nln42 on 2019-04-10 13:19:52 +0000 UTC_

**Describe the bug**

While trying to migrate from factcast-0.0.33 to factcast-0.1.0.M1 the startup of
factcast server using `factcast-store-pgsql` may fail if you do not increase database user rights to execute  Change Set `classpath:/db/changelog/factcast/issue329/createTokenStore.sql::issue329::usr` .

**FactCast Server**
Migration from factcast-0.0.33 to factcast-0.1.0.M1.
```
Exception encountered during context initialization - cancelling refresh attempt: 
org.springframework.beans.factory.BeanCreationException: 
Error creating bean with name 'liquibase' defined in class path resource [org/springframework/boot/autoconfigure/liquibase/LiquibaseAutoConfiguration$LiquibaseConfiguration.class]: 
Invocation of init method failed; nested exception is liquibase.exception.MigrationFailedException: 
Migration failed for change set classpath:/db/changelog/factcast/issue329/createTokenStore.sql::issue329::usr: 
Reason: liquibase.exception.DatabaseException: 
ERROR: permission denied to create extension "uuid-ossp" 
Hint: Must be superuser to create this extension. [Failed SQL: CREATE EXTENSION IF NOT EXISTS "uuid-ossp"]
```

Lets consider to give a hint in the migration guide.

_Copied from original issue: Mercateo/factcast#372_

**Comment by uweschaefer** [comment]
_From @mweirauch on 2019-04-10 16:10:42 +0000 UTC_

By accident stumbled upon the same issue last night on a docker-based 9.6.11-alpine where the dedicated user didn't have superuser rights.

The question is, does one need superuser rights under all circumstances, or can this be worked around? If so, this needs to be documented.

**Comment by uweschaefer** [comment]
_From @uweschaefer on 2019-04-10 18:57:09 +0000 UTC_

in order to avoid that, we could 

a) decompose the uuid and do not use the pg native uuid type. however, in theory, that seems like a bad patch, because pg knows best how to properly index UUIDs. Here using a composite key of two longs could be a pragmatic way to tackle this?
b) also demand the user to provide a pgsql with the extension already enabled but i am not sure that'd be a good practice either as it is not selfcontained, or 
c) document the need for the elevated privileges

your opinions?
@nln42 @mweirauch @OOlsen @StephanPraetsch @Mortinke ?

**Comment by uweschaefer** [comment]
_From @OOlsen on 2019-04-14 11:46:50 +0000 UTC_

Can't we do both b) and c) ? Make clear that there is a need for this extension. Either the user has to install it by himself or he has to give the  liquibase user the privileges. "CREATE EXTENSION IF NOT EXISTS" - will work in both cases.

**Comment by uweschaefer** [comment]
_From @uweschaefer on 2019-04-14 20:02:06 +0000 UTC_

would you mind adding this to the documentation? @OOlsen

**Comment by uweschaefer** [comment]
_From @OOlsen on 2019-04-20 14:43:57 +0000 UTC_

https://github.com/Mercateo/factcast/pull/388

**Comment by uweschaefer** [comment]
_From @uweschaefer on 2019-04-24 14:52:32 +0000 UTC_

closing this, as it will be resolved in #388
